libsplinter/src/service/scabbard/client/contract_registry.rs

```diff
@@ -0,0 +1,76 @@
+// Copyright 2020 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use sabre_sdk::protocol::payload::{
+    Action, CreateContractRegistryActionBuilder, DeleteContractRegistryActionBuilder, SabrePayload,
+    SabrePayloadBuilder, UpdateContractRegistryOwnersActionBuilder,
+};
+
+use super::Error;
+
+pub fn create_contract_registry_creation_payload(
+    name: &str,
+    owners: Vec<String>,
+) -> Result<SabrePayload, Error> {
+    let action = CreateContractRegistryActionBuilder::new()
+        .with_name(name.into())
+        .with_owners(owners)
+        .build()
+        .map_err(|err| {
+            Error::new_with_source("failed to build CreateContractRegistryAction", err.into())
```

Comment:
IMO, error messages should rarely contain source-code references because they will be meaningless to users if they are propagated up to them.

Reply:
I agree, but this error will only happen if a field is missing (which we can verify isn't), and if this error does occur, the only way to fix it would be by fixing the source code.

Reply:
Removed from this PR